handlers/product-move-api/src/main/scala/com/gu/productmove/zuora/Subscribe.scala

```diff
@@ -0,0 +1,88 @@
+package com.gu.productmove.zuora
+
+import com.gu.productmove.AwsS3
+import com.gu.productmove.GuStageLive.Stage
+import com.gu.productmove.zuora.{CaseId, ChargeOverrides, CreateSubscriptionResponse, ProductRatePlanChargeId, ProductRatePlanId, SubscribeToRatePlans, ZuoraAccountId}
+import com.gu.productmove.zuora.GetSubscription.GetSubscriptionResponse
+import com.gu.productmove.zuora.rest.ZuoraGet
+import sttp.capabilities.zio.ZioStreams
+import sttp.capabilities.{Effect, WebSockets}
+import sttp.client3.*
+import sttp.client3.httpclient.zio.{HttpClientZioBackend, SttpClient, send}
+import sttp.client3.ziojson.*
+import sttp.model.Uri
+import zio.json.*
+import zio.{Clock, IO, RIO, Task, UIO, URLayer, ZIO, ZLayer}
+
+import java.time.LocalDate
+
+trait Subscribe:
+  def create(zuoraAccountId: String, targetProductId: String): ZIO[Any, String, CreateSubscriptionResponse]
+
+object SubscribeLive :
+  val layer: URLayer[ZuoraGet, Subscribe] = ZLayer.fromFunction(SubscribeLive(_))
+
+private class SubscribeLive(zuoraGet: ZuoraGet) extends Subscribe:
+  override def create(zuoraAccountId: String, targetProductId: String): ZIO[Any, String, CreateSubscriptionResponse] = {
+    for {
+      subscribeRequest <- SubscribeRequest(zuoraAccountId, targetProductId)
+      response <- zuoraGet.post[SubscribeRequest, CreateSubscriptionResponse](uri"subscriptions", subscribeRequest)
+    } yield response
+  }
+
+object Subscribe {
+  def create(zuoraAccountId: String, targetProductId: String): ZIO[Subscribe, String, CreateSubscriptionResponse] =
+    ZIO.serviceWithZIO[Subscribe](_.create(zuoraAccountId, targetProductId))
+}
+
+case class ProductRatePlanId(value: String) extends AnyVal
+case class ProductRatePlanChargeId(value: String) extends AnyVal
+
+case class CreateSubscriptionResponse(subscriptionId: String)
+given JsonDecoder[CreateSubscriptionResponse] = DeriveJsonDecoder.gen[CreateSubscriptionResponse]
+
+case class ChargeOverrides(
+                            price: Option[Double],
+                            productRatePlanChargeId: String,
+                            triggerDate: Option[LocalDate],
+                            triggerEvent: Option[String]
+                          )
```

Comment:
intellij does this sometimes for some reason and I forget the setting to fix it, but I prefer to stick to 2 space indent.  This means if the `ChargeOverrides` class is renamed we don't have to reformat the other lines
```suggestion
case class ChargeOverrides(
  price: Option[Double],
  productRatePlanChargeId: String,
  triggerDate: Option[LocalDate],
  triggerEvent: Option[String]
)
```

Reply:
Ok, I'll have a google around this so it doesn't indent it lots on each re-format.